### ISSUE TITLE
Update binary compatibility plugin

### ISSUE THREAD

**Issue by CherryPerry** [badoo/Reaktive#608]
The latest version – 0.5.0.

**Comment by CherryPerry** [comment]
Blocked by https://github.com/Kotlin/binary-compatibility-validator/issues/47

**Comment by arkivanov** [comment]
Tried to update Kotlin version to `1.5.0` and the `binary-compatibility-validator` now dumps `internal` things. Could it be related?
See #610

**Comment by CherryPerry** [comment]
I think it might be.
You can try follow instructions in my comment in the related issue (disable `jvm` or `android` target and upgrade Kotlin version) and see what will happen.

**Comment by arkivanov** [comment]
If I understood correctly, I updated Kotlin to `1.5.0` and disabled `// setupAndroidTarget(project)` - internal things are still being dumped.

**Comment by CherryPerry** [comment]
And upgraded version of compatibility plugin to 0.5.0?

**Comment by arkivanov** [comment]
Looks like related to https://github.com/Kotlin/binary-compatibility-validator/issues/54 , we should to update try the new version of the plugin

**Comment by arkivanov** [comment]
Nah, then we will hit this issue :-) Looks like the update to `1.5.0` is also _kinda_ blocked by this task.

**Comment by CherryPerry** [comment]
I can make a patch to dump only `android` jar. Looks pretty simple [here](https://github.com/Kotlin/binary-compatibility-validator/blob/master/src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt#L62).

**Comment by arkivanov** [comment]
I can try, but it might be dangerous, we can miss something JVM-specific. Maybe better to keep internal things and just check manually per case? Not sure

**Comment by CherryPerry** [comment]
Actually in the current version of the plugin only `jvm` target is dumped [here](https://github.com/Kotlin/binary-compatibility-validator/blob/115f26ff22/src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt#L62).

**Comment by arkivanov** [comment]
Then perhaps we can disable `android`, not `jvm`?

**Comment by CherryPerry** [comment]
Anyway, can you check if dump works fine with validator 0.5.0 + kotlin 1.5.0 + disabled `android` sourceset on your branch? Meanwhile I will prepare separate PR with validator.

**Comment by arkivanov** [comment]
Kotlin `1.5.0` and plugin `0.5.0` work fine with the `android` target disabled.

**Comment by CherryPerry** [comment]
Fixed in https://github.com/Kotlin/binary-compatibility-validator/releases/tag/0.8.0-RC
Remove fork and use new version.
